ykrt/src/location.rs

```diff
@@ -100,36 +100,36 @@ impl Location {
     pub fn new() -> Self {
         // Locations start in the counting state with a count of 0.
         Self {
-            state: AtomicUsize::new(State::new().x),
+            state: AtomicUsize::new(LocationInternalState::new().x),
```

Comment:
Would `LocationInner` be more idiomatic of this repo? It's a bit shorter too :)

Reply:
Good idea. Fixed in https://github.com/ykjit/yk/pull/459/commits/5dd09b2be515f7c30386523b1a7c5fbbb3a8a957.